### ISSUE TITLE
Canot read properties of undefind (reading 'replace')

### ISSUE THREAD

**Issue by jirawatfreedom** [mschwarzmueller/nextjs-course-code#18]
error - components\events\event-item.js (17:36) @ EventItem
TypeError: Cannot read properties of undefined (reading 'replace')
  15 |     year: 'numeric',
  16 |   });
> 17 |   const formattedAddress = location.replace(', ', '\n');   
     |                                    ^
  18 |   const exploreLink = `/events/${id}`;
  19 | 
  20 |   return (

**Comment by fjerbi** [comment]
That Error message means that he couldn't access the location prop.
Are you passing `location ` as a prop?
you should have this :
` const { title, image, date, location, id } = props;`
In the beginning of your component.
and do not forget to pass `props `in your component.

**Comment by lukeryandev** [comment]
<img width="540" alt="image" src="https://user-images.githubusercontent.com/75271909/213259152-f7ec9bd3-ace0-48fe-badc-70f1861d7a81.png">

Same issue here, props passed in.

**Comment by lukeryandev** [comment]
I also added an optional chain, this is what displays when the formatting is skipped: 
EDIT: Same result when String(location) is applied.

const formattedAddress = location?.replace(", ", "\n");

<img width="745" alt="image" src="https://user-images.githubusercontent.com/75271909/213263855-5b5f410e-ea7c-48ad-beca-4b9aa4520f52.png">

**Comment by jasdeepdhillon13** [comment]
In some cases, this issue can be caused by not changing the firebase url in the api-utils.js file, which is set to point to the firebase by max. You should either replace the firebase url in the response variable to your own firebase url.

**Comment by Abdullah-encrypts** [comment]
> In some cases, this issue can be caused by not changing the firebase url in the api-utils.js file, which is set to point to the firebase by max. You should either replace the firebase url in the response variable to your own firebase url.

Yup when i was in that module i had the same issue and it was being caused by the firebase url
